state-sync/state-sync-v2/data-streaming-service/src/stream_engine.rs

```diff
@@ -1774,6 +1875,13 @@ impl SubscriptionStream {
     }
 }
 
+/// Bounds the given number by the specified min and max values, inclusive.
+/// If the number is less than the min, the min is returned. If the number is
+/// greater than the max, the max is returned. Otherwise, the number is returned.
+pub(crate) fn bound_by_range(number: u64, min: u64, max: u64) -> u64 {
```

Comment:
Should be equivalent to `number.clamp(min, max)`, in which case this utility isn't needed.

Reply:
🤦 Amazing, thanks @bchocho! I wondered if something like this existed, but didn't know it was called `clamp!`. Note: I've updated the implementation to use clamp, but still kept the wrapper method because I think it's slightly more readable. Plus, we reference it directly in a bunch of tests, and it feels a little cleaner to know that the implementation and tests are calling the same thing (as opposed to copy/pasting `clamp` everywhere). Does that make sense?

Reply:
IMO just using clamp makes sense. But I don't have a strong preference.

Readability is debatable :) I didn't know about it until recently either, but chatgpt tells me that C++, Python, and Javascript all have built-in `clamp` functions. (Java does not, which is probably why I didn't know about it)

Reply:
Gotcha! I'm inclined to leave it as-is (for now) and then maybe clean things up later 😄